File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left=0
        nleft=0
        answer=0
        for i in nums:
            if i==0:
                answer=max(answer,left+nleft)
                left=nleft
                nleft=0
            else:
                nleft+=1
        answer=max(answer,left+nleft)
        return answer if answer!=len(nums) else answer-1

        

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int nleft=0;
        int left=0;
        int answer=0;
        for(int i:nums){
            if(i==0){
                answer=Math.max(answer,left+nleft);
                left=nleft;
                nleft=0;

            }else{
                nleft++;
            }
       
        }
        answer=Math.max(answer,left+nleft);
        return answer!=nums.length?answer:answer-1;
    }
}